### PR TITLE
add option to append slash to link editable

### LINKED ISSUE #54: [Bug]: Cannot retain slash at the end of the URL on link editable under Documents
https://github.com/pimcore/pimcore/issues/14375

maybe we could add a symfony expression in the properties tab for string manipulations of the url.

### PR DESCRIPTION
**Pull request by cameronfromtorq**
Resolves #54 
image

add a slash if the option is checked and not if it isnt.

Note that I didnt go with a url modification function as that would require a backend call the system doesnt perform in this step (the url is generated on frontend only)

### CODE REVIEW

**Review by lukemacausland — APPROVED** [review]

**Review by kingjia90 — CHANGES_REQUESTED** [review]
Thank you!

Just took a quick view, found 2 issues. 

- While it works for the preview in admin interface, it doesn't appear on the frontend (public) 

Needs to add 
```
        if ($this->data['appendslash']){
            $url .= '/';
        }
```
around here, just before the `?` and `#`
https://github.com/TorqIT/pimcore/blob/88fddbcf708ed3aa7a502e04f1a6d1a0e9415f3c/models/Document/Editable/Link.php#L239-L241

- Needs also check the data type Link (data object), it seems not working properly at the moment
image
probably would be better to hide the checkbox there

File: models/Document/Editable/Link.php [hunk 1]
```diff
@@ -247,6 +246,10 @@ public function getHref()
             $url .= '#' . str_replace('#', '', $anchor);
         }
 
+        if ($this->data['appendslash']) {
```
Comment (kingjia90) [inline models/Document/Editable/Link.php]: needs to be a little bit earlier than `if (strlen($this->data['parameters'] ?? '') > 0) {` on line 239 as trailing slash should before `?somepar=123#anchor1`

**Review by lukemacausland — COMMENTED** [review]
Looks good, we need to make sure we disable / hide the append slash on dataobject link types or update the dataobject logic so that it saves / opens links as expected.

If we were to hide it, we can probably pass in the type to the openLinkEditPanel, and hide the append slash if its a dataobject type.

File: bundles/AdminBundle/Resources/public/js/pimcore/helpers.js [hunk 2]
```diff
@@ -1780,7 +1780,7 @@ pimcore.helpers.editmode = {};
 
 pimcore.helpers.editmode.openLinkEditPanel = function (data, callback) {
 
-
+    console.log(callback);
```
Comment (lukemacausland) [inline bundles/AdminBundle/Resources/public/js/pimcore/helpers.js]:
```suggestion
```

**Review by lukemacausland — APPROVED** [review]

**Pull request closed without merging**